Automattic/wp-calypso
# Store: Add the ability to reply to product reviews.

This PR adds the ability to reply to product reviews. The UI and behavior here matches normal comments management (see #17047 as well). It also adjusts the reply view to be sorted oldest -> newest like a thread.

<img width="771" alt="screen shot 2017-09-28 at 1 45 28 pm" src="https://user-images.githubusercontent.com/689165/30990139-03665752-a455-11e7-8f8e-53a79f75596a.png">

![cropped](https://user-images.githubusercontent.com/689165/30990174-1ec00f2a-a455-11e7-94c0-924731223138.png)

Closes #17047. Closes #17046.

To Test:
* Run `npm run test-client client/extensions/woocommerce/state` and make sure all tests pass.
* Go to `http://calypso.localhost:3000/store/reviews/:site` and find a review.
* Click into the comment textarea, and it should expand so you can begin typing your reply.
* Hit submit when you are ready. The reply should show up in the updated list.
* If you replied to a pending review, it should also be approved.
